e2e/test/scenarios/sharing/public-dashboard.cy.spec.js

```diff
@@ -245,4 +245,14 @@ describe("scenarios > public > dashboard", () => {
 
     assertDashboardFullWidth();
   });
+
+  it("should render when a filter passed with value starting from '0' (metabase#41483)", () => {
+    cy.get("@dashboardId").then(id => {
+      visitPublicDashboard(id, {
+        params: { number: "02" },
```

Comment:
could you use `text` as it's already existing?
![image](https://github.com/metabase/metabase/assets/1937582/f19ead96-c8c9-41ac-b936-15014e155f8d)

Then we should be able to query the widget value like so
```ts
    cy.url().should("include", "text=02");
    filterWidget().findByText("02").should("be.visible");
```